File: 326.power-of-three.js
/*
 * @lc app=leetcode id=326 lang=javascript
 *
 * [326] Power of Three
 */

// @lc code=start
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfThree = function (n) {
  while (n > 0 && n % 3 === 0) {
    n /= 3;
  }
  return n === 1;

  return n > 0 && Number.isInteger(Math.log10(n) / Math.log10(3));
};
// @lc code=end

File: 326.power-of-three.go
/*
 * @lc app=leetcode id=326 lang=golang
 *
 * [326] Power of Three
 */

package leetcode

// @lc code=start
func isPowerOfThree(n int) bool {
	for n > 0 && n%3 == 0 {
		n /= 3
	}
	return n == 1
}

// @lc code=end
